diff --git a/common/common.c b/common/common.c
index 86a618214..6583d5dd4 100644
--- a/common/common.c
+++ b/common/common.c
@@ -3116,6 +3116,13 @@ ssize_t select_write(serial_handler_t *fd, const void *buf, const size_t buflen,
  * communications media and/or vendor protocol.
  */
 static const char * search_paths_builtin[] = {
+	"@avahi@",
+	"@freeipmi@",
+	"@libusb@",
+	"@neon@",
+	"@libmodbus@",
+	"@netsnmp@",
+ "@libgpiod@",
 	/* Use the library path (and bitness) provided during ./configure first */
 	LIBDIR,
 	"/usr"LIBDIR,		/* Note: this can lead to bogus strings like */
