diff --git a/build.rs b/build.rs
index 48d1848..9aee5fa 100644
--- a/build.rs
+++ b/build.rs
@@ -1,16 +1,6 @@
 use std::process::Command;

 fn main() {
-    // commit hash
-    let git_describe = Command::new("git")
-        .arg("describe")
-        .arg("--always")
-        .arg("--dirty")
-        .arg("--tags")
-        .output()
-        .unwrap();
-
-    let git_describe = String::from_utf8(git_describe.stdout).unwrap();
     println!("cargo::rustc-env=GIT_DESCRIBE={git_describe}");

     // composite_templates

