diff --git a/loader/vulkan.pc.in b/loader/vulkan.pc.in
index 6a285f2..03ff061 100644
--- a/loader/vulkan.pc.in
+++ b/loader/vulkan.pc.in
@@ -6,5 +6,5 @@ includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR_PC@
 Name: Vulkan-Loader
 Description: Vulkan Loader
 Version: @VULKAN_LOADER_VERSION@
-Libs: -L${libdir} -lvulkan@VULKAN_LIB_SUFFIX@
-Cflags: -I${includedir}
+Libs: -L@CMAKE_INSTALL_LIBDIR@ -lvulkan@VULKAN_LIB_SUFFIX@
+Cflags: -I@CMAKE_INSTALL_INCLUDEDIR@
