diff --git a/src/gbm/backends/dri/gbm_dri.c b/src/gbm/backends/dri/gbm_dri.c
index 4e6b2b5..1210c1a 100644
--- a/src/gbm/backends/dri/gbm_dri.c
+++ b/src/gbm/backends/dri/gbm_dri.c
@@ -1223,7 +1223,7 @@ err_dri:
 
 struct gbm_backend gbm_dri_backend = {
    .v0.backend_version = GBM_BACKEND_ABI_VERSION,
-   .v0.backend_name = "dri",
+   .v0.backend_name = "dri_git",
    .v0.create_device = dri_device_create,
 };
 
diff --git a/src/gbm/main/backend.c b/src/gbm/main/backend.c
index 321c7e1..b88ae7c 100644
--- a/src/gbm/main/backend.c
+++ b/src/gbm/main/backend.c
@@ -162,7 +162,7 @@ _gbm_create_device(int fd)
       if (dev) return dev;
    }
 
-   return load_backend_by_name("dri", fd, true);
+   return load_backend_by_name("dri_git", fd, true);
 }
 
 void
