diff --git a/src/gbm/backends/dri/gbm_dri.c b/src/gbm/backends/dri/gbm_dri.c
index e8ca34d..1cf6f5f 100644
--- a/src/gbm/backends/dri/gbm_dri.c
+++ b/src/gbm/backends/dri/gbm_dri.c
@@ -1245,7 +1245,7 @@ err_dri:
 
 struct gbm_backend gbm_dri_backend = {
    .v0.backend_version = GBM_BACKEND_ABI_VERSION,
-   .v0.backend_name = "dri",
+   .v0.backend_name = "dri_git",
    .v0.create_device = dri_device_create,
 };
 
diff --git a/src/gbm/main/backend.c b/src/gbm/main/backend.c
index d5c7f3a..9db366f 100644
--- a/src/gbm/main/backend.c
+++ b/src/gbm/main/backend.c
@@ -155,7 +155,7 @@ static struct gbm_device *
 find_backend(const char *name, int fd)
 {
    struct gbm_device *dev = NULL;
-   dev = try_load_backend_by_name("dri", fd);
+   dev = try_load_backend_by_name("dri_git", fd);
 
    if (name && !dev) {
       dev = try_load_backend_by_name(name, fd);
